src/app/pages/News/News.tsx

```diff
@@ -1,98 +1,112 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import useFetch from '../../hooks/useFetch';
 import Card from '../../components/Card/Card';
 import Header from '../../components/Header/Header';
 import ToTop from '../../components/Icons/ToTop';
 import Navigation from '../../components/Navigation/Navigation';
 import styles from './News.module.css';
+import LoadingSpinner from '../../components/LoadingSpinner/LoadingSpinner';
 
-interface NewsMotd {
-  data: {
-    motds: [
-      {
-        id: string;
-        title: string;
-        body: string;
-        image: string;
-      }
-    ];
-  };
-}
-
-interface NewsStw {
-  data: {
-    messages: [
-      {
-        title: string;
-        body: string;
-        image: string;
-      }
-    ];
-  };
+interface newsFromAPI {
+  lang: string;
+  news: [
+    {
+      adspace: string;
+      body: string;
+      date: string;
+      id: string;
+      image: string;
+      live: boolean;
+      tabTitle: string;
+      title: string;
+      video: string | null;
+    }
+  ];
+  result: boolean;
+  show: number;
+  type: string;
 }
 
 export default function News(): JSX.Element {
-  const { data: battleRoyaleData } = useFetch<NewsMotd>(
-    'https://fortnite-api.com/v2/news/br'
-  );
-  const battleRoyaleNews = battleRoyaleData?.data.motds;
+  const [showToTop, setShowToTop] = useState(false);
 
-  const { data: creativeData } = useFetch<NewsMotd>(
-    'https://fortnite-api.com/v2/news/creative'
-  );
-  const creativeNews = creativeData?.data.motds;
+  const { data: battleRoyaleData, isLoading: battleRoyaleLoading } =
+    useFetch<newsFromAPI>('/api/news/?mode=br');
+  const battleRoyaleNews = battleRoyaleData?.news;
```

Comment:
This should be done in the backend. Only the relevant data should be returned to the frontend